### PR TITLE
Added reset password route and logic

### PR DESCRIPTION
**Pull request by adriangotca98**

### CODE REVIEW

File: controllers/authController.js [hunk 1]
```diff
@@ -4,6 +4,7 @@ const { createTkn } = require('../utils')
 exports.login = async (req, res) => {
   try {
     const user = await req.db.User.findOne({ ...req.body })
+    req.log.info(user)
```
Comment (procateodor) [inline controllers/authController.js]: I would remove this comment.

Comment (adriangotca98) [reply controllers/authController.js]: Ups, forgot it there. Will do by tomorrow morning 😅

Comment (adriangotca98) [reply controllers/authController.js]: resolved this last night

File: controllers/authController.js [hunk 2]
```diff
@@ -69,3 +69,34 @@ exports.register = async (req, res) => {
     })
   }
 }
+
+exports.resetPass = async (req, res) => {
+  try {
+    const userToSearchFor = { email: req.body.email, password: req.body.oldPass }
```
Comment (procateodor) [inline controllers/authController.js]: I would separate those in const {email, oldPassword, newPassword} = req.body

File: controllers/authController.js [hunk 3]
```diff
@@ -69,3 +69,34 @@ exports.register = async (req, res) => {
...
+    const newUserToSearchFor = { email: req.body.email, password: req.body.newPass }
+    const user = await req.db.User.findOne(userToSearchFor)
```
Comment (procateodor) [inline controllers/authController.js]: You should search first only for email, if user not exists return a `404` and user not found, after that search if the user that you searched for has the same password with oldPassword.

File: controllers/authController.js [hunk 4]
```diff
@@ -69,3 +69,34 @@ exports.register = async (req, res) => {
...
+    if (user === null) {
+      throw new Error('Wrong email or password')
+    }
+    const userChanged = await req.db.User.updateOne(userToSearchFor, { $set: { password: req.body.newPass } })
```
Comment (procateodor) [inline controllers/authController.js]: I think you could remove the `$set` object and put directly the object ```{password: newPassword}```.

File: controllers/authController.js [hunk 5]
```diff
@@ -69,3 +69,34 @@ exports.register = async (req, res) => {
...
+    if (userChanged.nModified !== 1) {
```
Comment (procateodor) [inline controllers/authController.js]: I think if mongo could not update the entry will throw an error so we shouldn't check that.

File: controllers/authController.js [hunk 6]
```diff
@@ -69,3 +69,34 @@ exports.register = async (req, res) => {
...
+      throw new Error('Database couldn\'t update the password!')
+    }
+    const newUser = await req.db.User.findOne(newUserToSearchFor)
```
Comment (procateodor) [inline controllers/authController.js]: we could get the user data from above in update.

File: controllers/authController.js [hunk 7]
```diff
@@ -69,3 +69,34 @@ exports.register = async (req, res) => {
...
+
+    const token = createTkn(
```
Comment (procateodor) [inline controllers/authController.js]: I think we shouldn't create a new token here.
just return a success status true.

File: controllers/authController.js [hunk 8]
```diff
@@ -69,3 +69,34 @@ exports.register = async (req, res) => {
...
+    )
+
+    return res.json({
+      success: true,
+      token
+    })
+  } catch (error) {
+    req.log.error(`Unable to change pass -> ${error}`)
+    return res.status(HttpStatus.INTERNAL_SERVER_ERROR).json({
+      success: false
```
Comment (procateodor) [inline controllers/authController.js]: pls add a message here like `Something bad happened!`.

File: routes/auth.js [hunk 9]
```diff
@@ -3,5 +3,6 @@ const { authController } = require('../controllers')
 
 router.post('/login', authController.login)
 router.post('/register', authController.register)
+router.post('/resetPass', authController.resetPass)
```
Comment (procateodor) [inline routes/auth.js]: use `reset_password` pls

**Review by procateodor — CHANGES_REQUESTED** [review]
Some minor changes.

ai mai facut niste schimbari in timp ce faceam review :(

**Comment by adriangotca98** [comment]
changes done after the feedback

**Comment by adriangotca98** [comment]
> Some minor changes.
> 
> ai mai facut niste schimbari in timp ce faceam review :(

Nu am dat scroll pana jos, sry

**Comment by adriangotca98** [comment]
[Create a new route for reset password.](https://trello.com/c/QMZnez94/7-create-a-new-route-for-reset-password)

**Review by procateodor — APPROVED** [review]

**Pull request merged**